ios/brave-ios/Sources/Brave/Frontend/Browser/Toolbars/UrlBar/TabLocationView.swift

```diff
@@ -524,39 +550,49 @@ class TabLocationView: UIView {
         // If we can't parse the origin and the URL can't be classified via AutoCompleteClassifier
         // the URL is likely a broken deceptive URL. Example: `about:blank#https://apple.com`
         if URLOrigin(url: url).url == nil && URIFixup.getURL(url.absoluteString) == nil {
-          urlDisplayLabel.isLeftToRight = true
-          urlDisplayLabel.text = url.scheme == "about" ? "about:blank" : ""
-        } else {
-          urlDisplayLabel.text = URLFormatter.formatURL(
-            URLOrigin(url: url).url?.absoluteString ?? url.absoluteString,
-            formatTypes: [
-              .omitDefaults, .trimAfterHost, .omitHTTPS, .omitTrivialSubdomains,
-            ],
-            unescapeOptions: .normal
-          )
-
-          // The direction the string will be rendered (this happens regardless of locale!!!)
-          // In a LTR environment, Arabic will always render RTL
-          // The dominant charset based on unicode's bidirection algorithm
-          // with strong L and strong R determines how a string will be rendered
-          let isRenderedLeftToRight = url.isRenderedLeftToRight
-
-          // Determine if the URL has BOTH LTR and RTL characters.
-          // Very likely a malicious URL, but not always!
-          // URLs like: m5155.xn--mgbaiqly6b2eg.xn--ngbc5azd/ are innocent
-          let isMixedCharset = !url.isUnidirectional
-
-          var isLTR = isRenderedLeftToRight && !isMixedCharset
-          if isMixedCharset {
-            // FORCE LTR - ETLD are always the right most portion after the dot.
-            // We will force render the URL in LTR mode, so we should force clip on the left (sub-domain).
-            // RTL rendered domains will clip from the right side (sub-domain) and the ETLD will be rendered on the left.
-            isLTR = true
+          if url.scheme == "about" {
+            url = URL(string: "about:blank")!
           }
+        }
 
-          urlDisplayLabel.isLeftToRight =
-            !["http", "https"].contains(url.scheme ?? "") || !isLTR
+        // Default is to format the FULL url
+        var urlString = url.absoluteString
+
+        // For the web schemes, format the origin instead of the full url
+        if ["http", "https"].contains(url.scheme) {
+          urlString = url.origin.url?.absoluteString ?? url.absoluteString
```

Comment:
May as well do an unwrap in the if instead.

e.g.

```swift
if url.isWebPage(), let originURL = url.origin.url {
  urlString = originURL
}
```

Reply:
Fixed.